hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/types/BucketObjectDBInfo.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.recon.api.types;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.hadoop.hdds.client.DefaultReplicationConfig;
+import org.apache.hadoop.hdds.protocol.StorageType;
+import org.apache.hadoop.ozone.om.helpers.BucketEncryptionKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.BucketLayout;
+import org.apache.hadoop.ozone.om.helpers.OmBucketInfo;
+
+/**
+ * Encapsulates the low level bucket info.
+ */
+public class BucketObjectDBInfo extends ObjectDBInfo {
```

Comment:
> Is it really a subclass of `ObjectDBInfo`? Or you just did it because they share some common properties?
> 
> It also doesn't follow the correct hierarchy model. Java doc comment says `VolumeObjectDBInfo` encapsulates the low level volume info, `BucketObjectDBInfo` encapsulates the low level bucket info and `ObjectDBInfo`Encapsulates the low level volume/bucket/dir info.
> 
> So hierarchy should be Volume -> Bucket -> Dir -> Key but according to your implementation Dir -> Volume, Bucket and Key. Which is wrong to me.
> 
> Please correct it. If you want to extract out common properties, create another class something `CommonPropertyClass` (Please come up with better name).
> 
> ```
> VolumeObjectDBInfo extands CommonPropertyClass
> BucketObjectDBInfo extands CommonPropertyClass
> ObjectDBInfo extands CommonPropertyClass
> KeyObjectDBInfo extands CommonPropertyClass
> ```

@hemantk-12 - As discussed, we have kept the name same as ObjectDBInfo. Pls re-review.